src/utils/promiseUtils.ts

```diff
@@ -24,3 +25,12 @@ export async function delay(ms: number, token?: vscode.CancellationToken): Promi
         }
     });
 }
+
+export async function getCancelPromise(token: vscode.CancellationToken, message?: string): Promise<never> {
+    return new Promise((resolve, reject) => {
+        const disposable = token.onCancellationRequested(() => {
+            disposable.dispose();
+            reject(new UserCancelledError(message));
```

Comment:
I guess if you squint, you can argue it was user action that cancelled the operation in the scenario addressed by this change

Reply:
My thoughts exactly! Though I could make that a generic.